quantities/Transformation.cs

```diff
@@ -0,0 +1,69 @@
+using Quantities.Numerics;
+
+namespace Quantities;
+
+public sealed class Transformation
```

Comment:
This class is the key enabler. It supports all relevant mathematical operations necessary to convert from one unit to the next.

I chose the form
$$f(x) = a/b \cdot x + c$$
So that inverting the function would not entail any divisions in the linear coefficients $a$ and $b$.

Basically, it records all necessary mathematical operations necessary for conversions.

Reply:
I don't think I understand what you mean with "the inverse function should not entail any divisions in the linear coefficients **_a_** and **_b_**".

If I'm not mistaken, the inverse function of the function above is:
$$f^{-1}(x) = (x-c)/(a/b)$$

I don't see any difference in expressing the function like:
$$f(x) = a \cdot x + b$$

Is it just a notation difference?

Reply:
After looking some more at the code, I think you wanted to avoid making the division _a/b_ as long as possible and that is why you worked with nominator and denominator. Is this right?

Reply:
You're exactly right. I want to avoid any actual arithmetic to compute the inverse until the last possible moment. (We can't avoid this for the offset.)

It's probably also a bit silly, but I though it "nicer" to enable inverses that remain exact. I.e. avoiding having to calculate $1/a$.